### PR TITLE
Trim admin's first name when validating

### LINKED ISSUE #13648: User Profile Name field not validated
### Required System information

- Node.js version: v16.13.2
- NPM version: 8.1.1
- Strapi version: v4.2.0
- Database: Sqlite
- Operating system: Windows

### Describe the bug

Name field under the User Profile section is marked as required but it can be easily bypassed using spaces.

### Steps to reproduce the behavior

1. Go to 'http://localhost:1337/admin/me'
2. Click on 'Name Field'
3. Press Space Bar 2-3 times
4. Save

### Expected behavior

When user input their name it should be trimmed to avoid spaces.

### Additional context

It saves exactly in database(eg. empty spaces string) what user enter in the field, there is no validation.

### PR DESCRIPTION
**Pull request by WalkingPizza**
### What does it do?

Fixes #13648.

### Why is it needed?

At the moment, the schema used to validate objects during admin creation/update does not trim `firstname`, allowing empty strings to be used as the admin's first name.

### How to test it?

Try creating/updating an admin using "   " as their first name.

### CODE REVIEW

**Review by ronronscelestes — COMMENTED** [review]
hi @WalkingPizza, thank you!
after discussing it with @Marc-Roig, we think it makes sense to trim `firstname` since it is a required field
though we should then probably apply the same logic to `Profile` + `Users` edit forms since `firstname` can be updated there too, what do you think about adding those changes to your PR? 🙂 

Users edit form:
image

Profile page form:
image

**Comment by WalkingPizza** [comment]
Hey @ronronscelestes, I am 99% sure I've already handled those scenarios as well already!

Let me try again to confirm.

**Comment by WalkingPizza** [comment]
Yup, I confirm they are already working.

image
image
image

**Review by ronronscelestes — APPROVED** [review]
Thank you for checking other forms containing `firstname` 😊
Looks good to me and works on my machine!
Do you want to review the backend side of the PR @Marc-Roig? 🙂

**Comment by Marc-Roig** [comment]
LGTM too! :) Everything is working in my machine. Nice job

**Review by Marc-Roig — APPROVED** [review]
